modules/vaos/app/services/eps/middleware/eps_logging.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module Eps
+  module Middleware
+    ##
+    # Faraday middleware that logs various semantically relevant attributes needed for debugging and audit purposes
+    #
+    class EpsLogging < Faraday::Middleware
+      STATSD_KEY_PREFIX = 'api.eps.response'
+
+      # #call
+      #
+      # Logs all outbound request / responses to VAMF api gateway as :info when success and :warn when fail
+      #
+      # Semantic logging tags:
+      # jti: The "jti" (JWT ID) claim provides a unique identifier for the JWT.
+      # status: The HTTP status returned from upstream.
+      # duration: The amount of time it took between request being made and response being received in seconds.
+      # url: The HTTP Method and URL invoked in the request.
+      #
+      # @param env [Faraday::Env] the request/response tree
+      # @return [Faraday::Env]
+      def call(env)
+        statsd_increment("#{STATSD_KEY_PREFIX}.total", env)
+        start_time = Time.current
+
+        @app.call(env).on_complete do |response_env|
+          if response_env.status.between?(200, 299)
+            log(:info, 'Eps service call succeeded!', log_tags(env, start_time, response_env))
+          elsif response_env.status == 400 # 400 error resp at times contain PII/PHI so we don't want the err msg logged
+            statsd_increment("#{STATSD_KEY_PREFIX}.fail", env)
+            log(:warn, 'Eps service call failed!', log_tags(env, start_time, response_env))
+          else
+            statsd_increment("#{STATSD_KEY_PREFIX}.fail", env)
+            log(:warn, 'Eps service call failed!', log_error_tags(env, start_time, response_env))
+          end
+        end
+      rescue Timeout::Error, Faraday::TimeoutError, Faraday::ConnectionFailed => e
+        statsd_increment("#{STATSD_KEY_PREFIX}.fail", env, e)
+        log(:warn, "Eps service call failed - #{e.message}", log_tags(env, start_time))
+        raise
+      end
+
+      private
+
+      def log_tags(env, start_time, response_env = nil)
+        anon_uri = VAOS::Anonymizers.anonymize_uri_icn(env.url)
+        {
+          jti: jti(env),
+          status: response_env&.status,
+          duration: Time.current - start_time,
+          # service_name: service_name || 'Eps Generic', # Need to figure out a clean way to do this with headers
```

Comment:
I think, we should be ok to use the `service_name`.  We have it defined in the `configuration` class [here](https://github.com/department-of-veterans-affairs/vets-api/blob/b9434ea6291d9b0342988425d65947b46ac278bc/modules/vaos/app/services/eps/configuration.rb#L11-L13).